File: haskell/src/Y2015.hs
module Y2015
    ( module Y2015Exports
    , puzzleMap, answerMap
    ) where

import Y2015.D01 as Y2015Exports
import Y2015.D02 as Y2015Exports
import Y2015.D03 as Y2015Exports
import Y2015.D04 as Y2015Exports
import Y2015.D05 as Y2015Exports
import Y2015.D06 as Y2015Exports
import Y2015.D07 as Y2015Exports
import Y2015.D08 as Y2015Exports
import Y2015.D09 as Y2015Exports
import Y2015.D10 as Y2015Exports
import Y2015.D11 as Y2015Exports
import Y2015.D12 as Y2015Exports
import Y2015.D13 as Y2015Exports
import Y2015.D14 as Y2015Exports
import Y2015.D15 as Y2015Exports
import Y2015.D16 as Y2015Exports
import Y2015.D17 as Y2015Exports
import Y2015.D18 as Y2015Exports
import Y2015.D19 as Y2015Exports
import Y2015.D20 as Y2015Exports
import Y2015.D21 as Y2015Exports
import Y2015.D22 as Y2015Exports
-- import Y2015.D23 as Y2015Exports
-- import Y2015.D24 as Y2015Exports
-- import Y2015.D25 as Y2015Exports

import Computation (Computation, Result(..))

import qualified Data.Map as M

puzzleMap :: M.Map Int Computation
puzzleMap = M.fromList . zip [1..] $
    [
      d01
    , d02
    , d03
    , d04
    , d05
    , d06
    , d07
    , d08
    , d09
    , d10
    , d11
    , d12
    , d13
    , d14
    , d15
    , d16
    , d17
    , d18
    , d19
    , d20
    , d21
    , d22
    -- , d23
    -- , d24
    -- , d25
    ]

answerMap :: M.Map Int [Result]
answerMap = M.fromList . fmap (fmap (Result <$>)) $ []

File: rust/src/y2015.rs
use crate::{Puzzle, Year, YearPuzzles};

pub const PUZZLES: YearPuzzles = YearPuzzles::new(YEAR, &PUZZLES_ARRAY);

const YEAR: Year = Year::Y2015;

const PUZZLES_ARRAY: [Option<Puzzle>; 25] = [
    Some(d01::PUZZLE),
    Some(d02::PUZZLE),
    Some(d03::PUZZLE),
    Some(d04::PUZZLE),
    Some(d05::PUZZLE),
    Some(d06::PUZZLE),
    Some(d07::PUZZLE),
    Some(d08::PUZZLE),
    Some(d09::PUZZLE),
    Some(d10::PUZZLE),
    Some(d11::PUZZLE),
    Some(d12::PUZZLE),
    Some(d13::PUZZLE),
    Some(d14::PUZZLE),
    Some(d15::PUZZLE),
    Some(d16::PUZZLE),
    Some(d17::PUZZLE),
    Some(d18::PUZZLE),
    Some(d19::PUZZLE),
    Some(d20::PUZZLE),
    Some(d21::PUZZLE),
    Some(d22::PUZZLE),
    Some(d23::PUZZLE),
    Some(d24::PUZZLE),
    Some(d25::PUZZLE),
];

mod d01;
mod d02;
mod d03;
mod d04;
mod d05;
mod d06;
mod d07;
mod d08;
mod d09;
mod d10;
mod d11;
mod d12;
mod d13;
mod d14;
mod d15;
mod d16;
mod d17;
mod d18;
mod d19;
mod d20;
mod d21;
mod d22;
mod d23;
mod d24;
mod d25;
